src/tools/quemap/lightmap.c

```diff
@@ -928,37 +1040,37 @@ void IndirectLighting(int32_t face_num) {
 	} else {
 		VectorCopy(plane->normal, normal);
 	}
+	
+	const vec_t *center = face_extents[face_num].center; // center of the face
 
-	const face_lighting_t *source_lighting = &face_lighting[face_num];
-
-	for (int32_t i = 0; i < source_lighting->num_samples; i++) {
+	face_lighting_t *fl = &face_lighting[face_num];
 
-		const vec_t *org = source_lighting->origins + i * 3;
-		const vec_t *sample = source_lighting->direct + i * 3;
-		const vec_t *direction = source_lighting->directions + i * 3;
+	for (int32_t i = 0; i < fl->num_samples; i++) { // calculate light for each sample
 
-		vec3_t color;
-		VectorCopy(sample, color);
+		vec_t *sample = fl->indirect + i * 3; // accumulate lighting here
+		vec_t *direction = fl->directions + i * 3; // accumulate direction here
 
-		const vec_t light = VectorLength(color) * material->hardness;
-		ColorNormalize(color, color);
+		byte pvs[(MAX_BSP_LEAFS + 7) / 8];
 
-		vec3_t reflect;
-		Reflect(direction, normal, reflect);
+		const vec_t *point = fl->origins + i * 3;
 
-		vec3_t end;
-		VectorMA(org, light, reflect, end);
+		if (texinfo->flags & SURF_PHONG) { // interpolated normal
+			SampleNormal(face, point, normal);
```

Comment:
I'm surprised you haven't torn SampleNormal a new one by now. It's clearly not correct.